File: calendario-outlook-backend/Models/Cita.cs
namespace CalendarioOutlook.Models
{
    public class Cita
    {
        public string Id { get; set; }
        public string Subject { get; set; }
        public Body Body { get; set; }
        public DateTimeDetails Start { get; set; }
        public DateTimeDetails End { get; set; }
        public Location Location { get; set; }
        public List<Attendee> Attendees { get; set; }
    }

    public class Body
    {
        public string ContentType { get; set; }
        public string Content { get; set; }
    }

    public class DateTimeDetails
    {
        public string DateTime { get; set; }
        public string TimeZone { get; set; }
    }

    public class Location
    {
        public string DisplayName { get; set; }
    }

    public class Attendee
    {
        public EmailAddress EmailAddress { get; set; }
        public string Type { get; set; }
    }

    public class EmailAddress
    {
        public string Address { get; set; }
        public string Name { get; set; }
    }

    public class CitaCreacion
    {
        public string Subject { get; set; }
        public Body Body { get; set; }
        public DateTimeDetails Start { get; set; }
        public DateTimeDetails End { get; set; }
        public Location Location { get; set; }
        public List<Attendee> Attendees { get; set; }
    }
}

File: calendario-outlook-front/src/app/cita.ts
export interface CitaCreacion {
  subject: string,
  body: {
    contentType: string,
    content: string
  },
  start: {
    dateTime: string,
    timeZone: string
  },
  end: {
    dateTime: string,
    timeZone: string
  },
  location: {
    displayName: string
  },
  attendees: Array<
    {
      emailAddress: {
        address: string,
        name: string
      },
      type: 'required' | 'optional'
    }>
}

export interface Cita {
  id: string,
  subject: string,
  body: {
    contentType: string,
    content: string
  },
  start: {
    dateTime: string,
    timeZone: string
  },
  end: {
    dateTime: string,
    timeZone: string
  },
  location: {
    displayName: string
  },
  attendees: Array<
    {
      emailAddress: {
        address: string,
        name: string
      },
      type: 'required' | 'optional'
    }>
}
